bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/CPU.java

```diff
@@ -3084,10 +3088,12 @@ public static boolean checkCast(BValue rhsValue, BType lhsType) {
         return checkCastByType(rhsType, lhsType);
     }
 
+    private static boolean checkCastToSameOrAnyType(BType rhsType, BType lhsType) {
```

Comment:
Shall we rename this to boolean related function name?

Reply:
Fixed with 1a378a5 as discussed.